### ISSUE TITLE
Network connectivity is always required

### ISSUE THREAD

**Issue by fotiDim** [spotify/ios-app-remote-sdk#8]
On every run of my app, even when it has already the necessary permissions for remote control and the track that I attempt to play is already downloaded in the Spotify app, starting that track requires network access. It would be nice if the permissions information was stored locally on the phone and didn't have to go through the network.

**Comment by hughrawlinson** [comment]
Hey @fotiDim, thanks for the feedback. I'll ask the team if they have any insights.

**Comment by kkarayannis** [comment]
Hi Foti! Our team is currently looking into this. I'll post an update here when this is resolved.

**Comment by fotiDim** [comment]
@kkarayannis nice to hear that someone is taking care of this. Thanks for the heads up!

**Comment by kkarayannis** [comment]
Hey Foti! This is now available in v1.2.0!

**Comment by fotiDim** [comment]
@kkarayannis I can only partially confirm it. I tried 3 scenarios and only 1 of them works.

- [X] If the app is already authenticated and I try while offline to play a downloaded song, this works.
- [] If the app is already authenticated and I try while offline to play a downloaded song *as radio*, this fails. The `playerAPI?.play` callback returns no error but a result instead. However the song is not played.
- [] If the app is not authenticated during that session and I try to authenticate, the Spotify app gives me this error. In a prior session the app was already authenticated. It seems that this is not persisted and it needs to go through the online validation process every single time.
![IMG_0032](https://user-images.githubusercontent.com/2326415/70389341-87243880-19be-11ea-88f5-1fba18dbb5c8.PNG)

**Comment by kkarayannis** [comment]
Hey Foti. Can you confirm the following:
- Your app has the latest SDK v1.2.0
- The Spotify app is version `8.5.30` or later.
- Your app authenticates once when online.
- Your app tries to authenticate when offline, no longer than 24 hours after it authenticated online and that the scopes weren't revoked manually in the meantime.

If that doesn't work, could you please try the NowPlayingView demo project in the SDK repo and see if that works for you?

**Comment by fotiDim** [comment]
@kkarayannis I have triple checked all the above. The `NowPlayingView` example works for me. I believe I understand now why the different behaviour. I am using `SPTSessionManager` authentication in my app while `NowPlayingView` uses built-in authorization. Is offline `SPTSessionManager` authentication not supported?

**Comment by eanntuan** [comment]
Hi @kkarayannis I'm following this thread, and was wondering if the tracks need to be downloaded to the user's device in order for offline playback to work (after first authenticating to Spotify with an internet connection). Thanks!

**Comment by kkarayannis** [comment]
@fotiDim Yes, you are correct that `SPTSessionManager` is not supported for the offline use case.

@eanntuan Yes, tracks need to be downloaded for offline use in the Spotify app beforehand, for offline playback to work.

**Comment by fotiDim** [comment]
@kkarayannis any plans to support SPTSessionManager? Otherwise it would be nice to state it in the readme.

**Comment by kkarayannis** [comment]
There are no plans to support SPTSessionManager. The docs have been updated to de-emphasize the use of SPTSessionManager.
